graphql/src/store/query.rs

```diff
@@ -68,6 +93,123 @@ pub(crate) fn build_query<'a>(
+        Some(r::Value::Int(n)) => {
+            let n = *n;
+            if n > 0 && n <= (max_first as i64) {
+                n as u32
+            } else {
+                return Err(QueryExecutionError::RangeArgumentsError(
+                    "first", max_first, n,
+                ));
+            }
+        }
+        Some(r::Value::Null) | None => 100,
+        _ => unreachable!("first is an Int with a default value"),
+    };
+
+    let last = match field.argument_value("last") {
+        Some(r::Value::Int(n)) => {
+            let n = *n;
+            if n >= 0 && n <= (max_skip as i64) {
+                n as u32
+            } else {
+                return Err(QueryExecutionError::RangeArgumentsError(
+                    "last", max_skip, n,
+                ));
+            }
+        }
+        Some(r::Value::Null) | None => 0,
+        _ => unreachable!("last is an Int with a default value"),
+    };
+
+    let after = match field.argument_value("after") {
+        Some(r::Value::String(s)) => Some(s.clone()),
+        Some(r::Value::Null) | None => None,
+        _ => unreachable!("after is a String"),
+    };
+
+    let before = match field.argument_value("before") {
+        Some(r::Value::String(s)) => Some(s.clone()),
+        Some(r::Value::Null) | None => None,
+        _ => unreachable!("before is a String"),
+    };
+
+    // Validate compatible input arguments
+    if first > 0 && last > 0 {
+        return Err(QueryExecutionError::InvalidCursorOptions(
+            "only one of \"first\" and \"last\" may be provided".to_string(),
+        ));
+    } else if before.is_some() && after.is_some() {
+        return Err(QueryExecutionError::InvalidCursorOptions(
+            "only one of \"before\" and \"after\" may be provided".to_string(),
+        ));
+    } else if first > 0 && before.is_some() {
+        return Err(QueryExecutionError::InvalidCursorOptions(
+            "\"first\" may only be used with \"after\"".to_string(),
+        ));
+    } else if last > 0 && after.is_some() {
+        return Err(QueryExecutionError::InvalidCursorOptions(
+            "\"last\" may only be used with \"before\"".to_string(),
+        ));
+    }
+
+    // If `after` exists then create entity filter
+    let after_sql: Option<EntityFilter> = match after {
+        Some(s) => {
+            let decode_base64 = base64::decode(s.clone());
+
+            match decode_base64 {
+                Ok(decoded) => {
+                    let decoded_value = String::from_utf8(decoded);
+
+                    match decoded_value {
+                        Ok(val) => {
+                            // We encode the cursors using `:` as a separator
+                            // split by `:` and getting the first will give us the ID
+                            // these seems to be `"` around the ID so we need to remove them
+                            let id = val.split(":").collect::<Vec<&str>>()[0].replace("\"", "");
```

Comment:
Did you ever figure out where the extra `\"` are coming from? We should really get to the bottom of this before merging.